### PR TITLE
Update FulfillRequisitionRequest and ProtocolConfig for ShuffleBasedSecretSharing protocol

### PR DESCRIPTION
**Pull request by renjiezh**
For shuffle based secret sharing protocol, requisitions are fulfilled by EDPs with either chunks of data (a blob) or a seed. The seed can be expanded into a deterministic blob with the same PRNG by workers.
The change is back-compatible.

Also added ProtocolConfig in FulfillRequisitionRequest for this issue(https://github.com/world-federation-of-advertisers/cross-media-measurement/issues/1329).

### CODE REVIEW

**Comment by wfa-reviewable** [comment]
This change is [Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191)

**Review by kungfucraig — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @ple13, @renjiezh, and @SanjayVas)

___
*[`src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto` line 35 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-Njhtv_V5MgZ-P7ssyB6:-Njhtv_V5MgZ-P7ssyB7:b83z54x) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/bbc98005f9f67c01458b92d586067ed75c5ec207/src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto#L35)):*
> ```proto
> 
> // Request message for `FulfillRequisition` method.
> message FulfillRequisitionRequest {
> ```

What do you think about adding a field for the protocol config being used?

**Review by renjiezh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @kungfucraig, @ple13, and @SanjayVas)

___
*[`src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto` line 35 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-Njhtv_V5MgZ-P7ssyB6:-NjhwSE89zG0s87MXP4j:b-lyzphc) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/bbc98005f9f67c01458b92d586067ed75c5ec207/src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto#L35)):*
<details><summary><i>Previously, kungfucraig (Craig Wright) wrote…</i></summary><blockquote>

What do you think about adding a field for the protocol config being used?
</blockquote></details>

Thanks.
Covered that in this PR.

**Review by renjiezh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @kungfucraig, @ple13, and @SanjayVas)

___
*[`src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto` line 211 at r4](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjidhEj6tmf9Cix6HPY:-NjidhEj6tmf9Cix6HPZ:b4rj37j) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/e751973797d46e7400de06e5c2beff9fcc04657e/src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto#L211)):*
> ```proto
>     // The maximum frequency to reveal in the histogram. This is also the EDP
>     // frequency cap.
>     int32 maximum_frequency = 2;
> ```

LLv2's protocol config deprecated this field as it is in measurement_spec already. However, for ShuffleBasedSecretSharing we want to validate ProtocolConfig. Do we want to also compare this field?
@kungfucraig @SanjayVas

**Review by renjiezh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191)* status: 1 of 3 files reviewed, 2 unresolved discussions (waiting on @kungfucraig and @ple13)

___
*[`src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto` line 211 at r4](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjidhEj6tmf9Cix6HPY:-NjjcTZd6UsLEti_tJ05:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/e751973797d46e7400de06e5c2beff9fcc04657e/src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto#L211)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

We shouldn't include this field here in ProtocolConfig for the same reason we dropped it from LLv2: it's specified in MeasurementSpec. Drop this field.
</blockquote></details>

Done.

**Review by ple13 — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r3, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191)* status: all files reviewed, 4 unresolved discussions (waiting on @kungfucraig and @renjiezh)

___
*[`src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto` line 69 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjlA7IU0LVaW8n9K8uO:-NjlA7IU0LVaW8n9K8uP:bribh7v) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/511217674a766f73712251cb36fcb7d43ca0b75e/src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto#L69)):*
> ```proto
>     message ShuffleBasedSecretSharing {
>       // The seed can be expanded into a deterministic blob using the same PRNG.
>       bytes seed = 1;
> ```

Does `seed` stand for both blob/seed?
___
*[`src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto` line 72 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjlBVUa1am5vemFSZ_x:-NjlBVUa1am5vemFSZ_y:b-ti41dl) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/511217674a766f73712251cb36fcb7d43ca0b75e/src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto#L72)):*
> ```proto
>     }
>     // Protocol specified values.
>     oneof protocol {
> ```

Why do we use oneof when there's only 1 option?

**Review by kungfucraig — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r3, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191)* status: all files reviewed, 8 unresolved discussions (waiting on @renjiezh and @SanjayVas)

___
*[`src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto` line 205 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjmppPD-IiZCJG_kbXQ:-NjmppPD-IiZCJG_kbXR:b-xuji3k) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/511217674a766f73712251cb36fcb7d43ca0b75e/src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto#L205)):*
> ```proto
> 
>   // Configuration for the Shuffle Based Secret Sharing Protocol.
>   message ShuffleBasedSecretSharing {
> ```

Can we call this: HonestMajoritySecretShareShuffle?

I think having HM in the name is helpful.
___
*[`src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto` line 286 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjmrXuc4f6MFPSrDfcJ:-NjmrXud7Q7yBp_jNoCo:b-p7qdb7) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/511217674a766f73712251cb36fcb7d43ca0b75e/src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto#L286)):*
> ```proto
> message SecretSharingSketchParams {
>   // The size of the sketch.
>   int64 sketch_size = 1;
> ```

Should we call this register count? 
___
*[`src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto` line 290 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjmpXU4D4IPtmzq6g6E:-NjmpXU4D4IPtmzq6g6F:b12cofm) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/511217674a766f73712251cb36fcb7d43ca0b75e/src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto#L290)):*
> ```proto
>   // Length of each item in bits. #_EDPs * max_frequency should be no more than
>   // 2^bit_per_item.
>   int32 bit_per_item = 2;
> ```

Let's make this "bytes_per_item," and in the future if we need bit level granularity we can make a oneof that has the option of bits or bytes. It will keep the implementation simpler to assume bytes.

@ple13 do we need to specify Z_p in this structure?
___
*[`src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto` line 62 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjmsA6Q9rjup31n-9kl:-NjmsA6Q9rjup31n-9km:bmd2bow) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/511217674a766f73712251cb36fcb7d43ca0b75e/src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto#L62)):*
> ```proto
>     // The protocol config of the Computation. This is used to validate that
>     // EDPs and the MPC are using the same protocol config.
>     // Required for ShuffleBasedSecretSharing protocol.
> ```

Should we say "not required for LLv2 protocols?" Then we can assume that going forward we'll always expect this.

**Review by ple13 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191)* status: 1 of 3 files reviewed, 8 unresolved discussions (waiting on @kungfucraig, @renjiezh, and @SanjayVas)

___
*[`src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto` line 290 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjmpXU4D4IPtmzq6g6E:-Njn3YVTEowuGyK7GIJF:b621ybe) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/511217674a766f73712251cb36fcb7d43ca0b75e/src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto#L290)):*
<details><summary><i>Previously, kungfucraig (Craig Wright) wrote…</i></summary><blockquote>

Let's make this "bytes_per_item," and in the future if we need bit level granularity we can make a oneof that has the option of bits or bytes. It will keep the implementation simpler to assume bytes.

@ple13 do we need to specify Z_p in this structure?
</blockquote></details>

For the frequency protocol, we use the ring 2^{byte_per_item*8}. Later on, if we want a separate protocol for reach, we will need to use a prime and need to specify one.

**Review by renjiezh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191)* status: 1 of 3 files reviewed, 8 unresolved discussions (waiting on @kungfucraig, @ple13, and @SanjayVas)

___
*[`src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto` line 205 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjmppPD-IiZCJG_kbXQ:-NjnEc1n4wRjIOHE4EB6:bxn9b05) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/511217674a766f73712251cb36fcb7d43ca0b75e/src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto#L205)):*
<details><summary><i>Previously, kungfucraig (Craig Wright) wrote…</i></summary><blockquote>

Can we call this: HonestMajoritySecretShareShuffle?

I think having HM in the name is helpful.
</blockquote></details>

Done.
Renamed to `HonesMajorityShareShuffle`. Internal APIs will use the same name.
___
*[`src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto` line 286 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjmrXuc4f6MFPSrDfcJ:-Njn2GK-DCaUo4_-GSqo:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/511217674a766f73712251cb36fcb7d43ca0b75e/src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto#L286)):*
<details><summary><i>Previously, kungfucraig (Craig Wright) wrote…</i></summary><blockquote>

Should we call this register count?
</blockquote></details>

Done.
___
*[`src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto` line 290 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjmpXU4D4IPtmzq6g6E:-Njn2Hl-1iYLsN208eWI:bs7wumb) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/511217674a766f73712251cb36fcb7d43ca0b75e/src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto#L290)):*
<details><summary><i>Previously, ple13 (Phi) wrote…</i></summary><blockquote>

For the frequency protocol, we use the ring 2^{byte_per_item*8}. Later on, if we want a separate protocol for reach, we will need to use a prime and need to specify one.
</blockquote></details>

Done.
___
*[`src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto` line 62 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjmsA6Q9rjup31n-9kl:-Njn2f3F2IZ-uRjYI5tN:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/511217674a766f73712251cb36fcb7d43ca0b75e/src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto#L62)):*
<details><summary><i>Previously, kungfucraig (Craig Wright) wrote…</i></summary><blockquote>

Should we say "not required for LLv2 protocols?" Then we can assume that going forward we'll always expect this.
</blockquote></details>

Done.
___
*[`src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto` line 69 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjlA7IU0LVaW8n9K8uO:-Njn2ler2DrZ01mG7x4I:bmxq0d) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/511217674a766f73712251cb36fcb7d43ca0b75e/src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto#L69)):*
<details><summary><i>Previously, ple13 (Phi) wrote…</i></summary><blockquote>

Does `seed` stand for both blob/seed?
</blockquote></details>

Added comment to make it clearer.
The requisition data is either a seed from the header or blob of chunks .
___
*[`src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto` line 72 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjlBVUa1am5vemFSZ_x:-Njn37zp2hE_8YHzZANd:b-drt4we) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/511217674a766f73712251cb36fcb7d43ca0b75e/src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto#L72)):*
<details><summary><i>Previously, ple13 (Phi) wrote…</i></summary><blockquote>

Why do we use oneof when there's only 1 option?
</blockquote></details>

Each requisition could contains protocol specified values, e.g,. `seed`.  As we might have multiple protocols, oneof fits the usage.
Ideally, each fulfillment can have both values in header and blobs in following chunks. The protocol field should provides information about both of them. However, we omitted the info regarding chunks previously.

**Review by kungfucraig — COMMENTED** [review]
> Reviewed 1 of 2 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191)* status: 2 of 3 files reviewed, 3 unresolved discussions (waiting on @ple13 and @SanjayVas)

___
*[`src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto` line 290 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjmpXU4D4IPtmzq6g6E:-NjnHkt07CJLzPm7OqmY:b-2326u5) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/511217674a766f73712251cb36fcb7d43ca0b75e/src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto#L290)):*
<details><summary><i>Previously, renjiezh wrote…</i></summary><blockquote>

Done.
</blockquote></details>

nit: bytes_per_item

**Review by SanjayVas — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r9, 1 of 1 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191)* status: all files reviewed, 7 unresolved discussions (waiting on @ple13 and @renjiezh)

___
*[`src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto` line 288 at r10](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjnIeKbA-drOaPw_Dbz:-NjnIeKc27VOaJcytpBA:b33bwiz) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/0c55f4da7ab0283cfced7d760227d4ad7573c5b5/src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto#L288)):*
> ```proto
>   int64 register_count = 1;
> 
>   // Length of each item in bytes. #_EDPs * max_frequency should be no more than
> ```

What is "item" in this case? Is this a known term w.r.t sketch creation?



_Code quote:_
```proto
item
```
___
*[`src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto` line 289 at r10](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjnNSEiCz8LHRO9WcBi:-NjnNSEiCz8LHRO9WcBj:bmniyma) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/0c55f4da7ab0283cfced7d760227d4ad7573c5b5/src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto#L289)):*
> ```proto
> 
>   // Length of each item in bytes. #_EDPs * max_frequency should be no more than
>   // 2^(bytes_per_item*8).
> ```

Reference specific messages and fields. The abbreviation "EDP" is not used in the API.



_Suggestion:_
```proto
  // Length of each item in bytes.
  //
  // The product of `maximum_frequency` and the `nonce_hashes` count from the `MeasurementSpec` should be no more than 2 ^ (`bytes_per_item` * 8).
```
___
*[`src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto` line 69 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjlA7IU0LVaW8n9K8uO:-NjnJX-o1Q_-X6-bNHrs:b-hf7unb) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/511217674a766f73712251cb36fcb7d43ca0b75e/src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto#L69)):*
<details><summary><i>Previously, renjiezh wrote…</i></summary><blockquote>

Added comment to make it clearer.
The requisition data is either a seed from the header or blob of chunks .
</blockquote></details>

To clarify, this means that you'll either have a seed specified in the header and no chunks *or* no seed specified in the header and a set of chunks, correct?
___
*[`src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto` line 63 at r10](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjnJ2-B1EGPpb9NRv5k:-NjnJ2-B1EGPpb9NRv5l:bjqulnx) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/0c55f4da7ab0283cfced7d760227d4ad7573c5b5/src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto#L63)):*
> ```proto
>     // EDPs and the MPC are using the same protocol config.
>     //
>     // NOT required for LiquidLegionV2 protocols.
> ```

This is implying it will never be required for those protocols, where I thought we wanted to make this field unconditionally required in the future.

Maybe rephrase this to make it clear that this is required for the HMShareShuffle protocol and may eventually be required for all protocols.



_Code quote:_
```proto
NOT required for LiquidLegionV2 protocols
```
___
*[`src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto` line 72 at r10](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjnKdEu7lIpjt2Oj7om:-NjnKdEv1pvvxR2CbZyG:b88y4of) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/0c55f4da7ab0283cfced7d760227d4ad7573c5b5/src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto#L72)):*
> ```proto
>       //
>       // If the seed is not specified or empty, it means the requisition is
>       // fulfilled by a blob of chunks.
> ```

The first line of each of each message/field doc should be a [summary fragment](https://developer.android.com/kotlin/style-guide#documentation)[^1] which should describe the message/field itself.

Leave out all mention of a blob here, since that's not part of the API. Also note that for singular proto3 scalar fields, "not specified" is the same as "set to default value". Therefore there's no reason to include "or empty" for a `bytes` field.

[^1]: This is from the Kotlin style guide, but applies to all languages



_Suggestion:_
```proto
    // Header for the HonestMajorityShareShuffle protocol.
    message HonestMajorityShareShuffle {
      // <explanation of what the seed is and where it comes from, e.g. if it's pre-shared across DataProviders>
      //
      // If specified then there should be no subsequent requests with `body_chunk`s.
```
___
*[`src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto` line 92 at r10](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjnMAEG3p19FLrSHAQN:-NjnMAEH9JF8VO_4WhkM:bvwharl) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/0c55f4da7ab0283cfced7d760227d4ad7573c5b5/src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto#L92)):*
> ```proto
>     // the combined `Duchy` ElGamal public keys. The only alignment requirement
>     // is by bytes: a chunk might begin or end in the middle of a single
>     // register.
> ```

Need to update this comment as I believe it differs for the new protocol.



_Code quote:_
```proto
    // The format of the data depends on the corresponding `MeasurementSpec`. If
    // the `Requisition` is for an encrypted sketch, this is the register
    // data as documented in the `Sketch` message (sketch.proto) encrypted using
    // the combined `Duchy` ElGamal public keys. The only alignment requirement
    // is by bytes: a chunk might begin or end in the middle of a single
    // register.
```

**Review by renjiezh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191)* status: 1 of 3 files reviewed, 7 unresolved discussions (waiting on @ple13 and @SanjayVas)

___
*[`src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto` line 288 at r10](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjnIeKbA-drOaPw_Dbz:-NjniVCCCu-hyCEmWxqD:bdswzb7) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/0c55f4da7ab0283cfced7d760227d4ad7573c5b5/src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto#L288)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

What is "item" in this case? Is this a known term w.r.t sketch creation?
</blockquote></details>

Renamed to `register`
___
*[`src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto` line 289 at r10](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjnNSEiCz8LHRO9WcBi:-NjntHvOC50Qu_6j7Fpd:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/0c55f4da7ab0283cfced7d760227d4ad7573c5b5/src/main/proto/wfa/measurement/api/v2alpha/protocol_config.proto#L289)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

Reference specific messages and fields. The abbreviation "EDP" is not used in the API.
</blockquote></details>

Done.
___
*[`src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto` line 69 at r7](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjlA7IU0LVaW8n9K8uO:-Njnko7w8cQ2xIYIvND5:b-4w3a2s) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/511217674a766f73712251cb36fcb7d43ca0b75e/src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto#L69)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

To clarify, this means that you'll either have a seed specified in the header and no chunks *or* no seed specified in the header and a set of chunks, correct?
</blockquote></details>

For our new protocol, yes.
But it supports both appearances of `seed` (along with other values if needed) and chunks. 
___
*[`src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto` line 63 at r10](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjnJ2-B1EGPpb9NRv5k:-NjnsobvD2RntcYl1k5b:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/0c55f4da7ab0283cfced7d760227d4ad7573c5b5/src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto#L63)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

This is implying it will never be required for those protocols, where I thought we wanted to make this field unconditionally required in the future.

Maybe rephrase this to make it clear that this is required for the HMShareShuffle protocol and may eventually be required for all protocols.
</blockquote></details>

Done.
___
*[`src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto` line 72 at r10](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjnKdEu7lIpjt2Oj7om:-NjnspOh5HQ8MhpDw1gF:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/0c55f4da7ab0283cfced7d760227d4ad7573c5b5/src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto#L72)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

The first line of each of each message/field doc should be a [summary fragment](https://developer.android.com/kotlin/style-guide#documentation)[^1] which should describe the message/field itself.

Leave out all mention of a blob here, since that's not part of the API. Also note that for singular proto3 scalar fields, "not specified" is the same as "set to default value". Therefore there's no reason to include "or empty" for a `bytes` field.

[^1]: This is from the Kotlin style guide, but applies to all languages
</blockquote></details>

Done.
___
*[`src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto` line 92 at r10](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjnMAEG3p19FLrSHAQN:-NjntJJ89H_6910v4z0I:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/0c55f4da7ab0283cfced7d760227d4ad7573c5b5/src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto#L92)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

Need to update this comment as I believe it differs for the new protocol.
</blockquote></details>

Done.

**Review by SanjayVas — APPROVED** [review]
> Reviewed 2 of 2 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191)* status: all files reviewed, 3 unresolved discussions (waiting on @ple13 and @renjiezh)

___
*[`src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto` line 72 at r10](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjnKdEu7lIpjt2Oj7om:-NjoFAJVD6BfXkWqzxhd:b5qlxn9) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/0c55f4da7ab0283cfced7d760227d4ad7573c5b5/src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto#L72)):*
nit

From the linked style guide:

> This is a fragment–a noun phrase or verb phrase, not a complete sentence. It does not begin with "``A `Foo` is a...``" 

This means the the summary fragment shouldn't read "The seed is..." but rather e.g. "A random seed which is..."
___
*[`src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto` line 63 at r11](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-NjoExxGCwB_A1NlR1-o:-NjoExxGCwB_A1NlR1-p:b-le35f8) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/d3b905212ac33a206cbc2ed031f4b2bab63d1d35/src/main/proto/wfa/measurement/api/v2alpha/requisition_fulfillment_service.proto#L63)):*
> ```proto
>     // EDPs and the MPC are using the same protocol config.
>     //
>     // Required for HonestMajorityShareShuffle protocol. LiquidLegionsV2
> ```

nit: indicate that it will be required for all protocols. We don't need to enumerate them, and the Direct protocol doesn't involve calling this method.



_Code quote:_
```proto
 LiquidLegionsV2
```

**Review by ple13 — COMMENTED** [review]
> Reviewed 2 of 2 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191)* status: all files reviewed, 3 unresolved discussions (waiting on @renjiezh)

**Review by ple13 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191#-:-NjoJSz56gTLJM_6QE2Y:bnfp4nl)* status: all files reviewed, 3 unresolved discussions (waiting on @renjiezh)

**Review by ple13 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191)* status: all files reviewed, 1 unresolved discussion (waiting on @renjiezh)

**Review by stevenwarejones — APPROVED** [review]
> Reviewed 1 of 2 files at r3, 1 of 2 files at r11, 1 of 1 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @renjiezh)

**Review by kungfucraig — APPROVED** [review]
> Reviewed 1 of 2 files at r11, 1 of 1 files at r12.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/191)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @renjiezh)
